Install button provides incorrect directory path

[Here's an example:](https://portal.azure.com/#view/Microsoft_Azure_Monitoring_Logs/LogsBlade/resourceId/%2Fsubscriptions%2Fa4af4729-21ee-4e0a-953c-33cc9db5295c%2Fresourcegroups%2Fvscode-extension%2Fproviders%2Fmicrosoft.insights%2Fcomponents%2Fvscode-extension-analytics/source/AIExtension.SearchV2/query/union%20customEvents%20%7C%20where%20timestamp%20%3E%3D%20datetime(%222022-10-22T02%3A15%3A10.248Z%22)%20and%20timestamp%20%3C%3D%20(datetime(%222022-10-22T02%3A15%3A10.248Z%22))%20and%20customDimensions%5B%22appmap.cli.sessionId%22%5D%20contains%20%22cf79b925a8c751cb1fdccd1cbd248d4ea28095a2e9f70ddd3a7c4fd9ff35aaba%22%7C%20extend%20logLen%3Dstrlen(customDimensions%5B%22appmap.cli.log%22%5D)%2Clog%3DcustomDimensions%5B%22appmap.cli.log%22%5D%7C%20project-reorder%20%20timestamp%2Cname%2Cuser_Id%2ClogLen%2Clog)
```
C:\Users\lopez\AppData\Local\npm-cache\_npx\3007b677135bbf3c\node_modules\@appland\appmap\built\cli.js install -d c:UserslopezOneDriveDocumentosGitHubFeedback
```

[Another example:](https://portal.azure.com/#view/Microsoft_Azure_Monitoring_Logs/LogsBlade/resourceId/%2Fsubscriptions%2Fa4af4729-21ee-4e0a-953c-33cc9db5295c%2Fresourcegroups%2Fvscode-extension%2Fproviders%2Fmicrosoft.insights%2Fcomponents%2Fvscode-extension-analytics/source/AIExtension.SearchV2/query/union%20customEvents%20%7C%20where%20timestamp%20%3E%3D%20datetime(%222022-10-22T15%3A43%3A07.121Z%22)%20and%20timestamp%20%3C%3D%20(datetime(%222022-10-22T15%3A43%3A07.121Z%22))%20and%20customDimensions%5B%22appmap.cli.sessionId%22%5D%20contains%20%22c8c4905be6d18687b703233d3dba5c85c74a95544146a3d4b47c5623fb542361%22%7C%20extend%20logLen%3Dstrlen(customDimensions%5B%22appmap.cli.log%22%5D)%2Clog%3DcustomDimensions%5B%22appmap.cli.log%22%5D%7C%20project-reorder%20%20timestamp%2Cname%2Cuser_Id%2ClogLen%2Clog)
```
Running command: `npm install --saveDev @appland/appmap-agent-js@latest`

npm ERR! code ENOTDIR
npm ERR! syscall mkdir
npm ERR! path g:\Mi unidad\Cursos\Programacion\AVATAR\node_modules\http-proxy-agent\node_modules
npm ERR! errno -4052
npm ERR! ENOTDIR: not a directory, mkdir 'g:\Mi unidad\Cursos\Programacion\AVATAR\node_modules\http-proxy-agent\node_modules'
```

[Another log](https://portal.azure.com/#view/Microsoft_Azure_Monitoring_Logs/LogsBlade/resourceId/%2Fsubscriptions%2Fa4af4729-21ee-4e0a-953c-33cc9db5295c%2Fresourcegroups%2Fvscode-extension%2Fproviders%2Fmicrosoft.insights%2Fcomponents%2Fvscode-extension-analytics/source/AIExtension.SearchV2/query/union%20customEvents%20%7C%20where%20timestamp%20%3E%3D%20datetime(%222022-10-24T07%3A31%3A34.982Z%22)%20and%20timestamp%20%3C%3D%20(datetime(%222022-10-24T07%3A31%3A34.982Z%22))%20and%20customDimensions%5B%22appmap.cli.sessionId%22%5D%20contains%20%225a68a174a8fba29b100f158df52e4e7765834c3f13d4e4ee2713e0132dafcc7d%22%7C%20extend%20logLen%3Dstrlen(customDimensions%5B%22appmap.cli.log%22%5D)%2Clog%3DcustomDimensions%5B%22appmap.cli.log%22%5D%7C%20project-reorder%20%20timestamp%2Cname%2Cuser_Id%2ClogLen%2Clog)

--- a/src/commands/installAgent.ts
+++ b/src/commands/installAgent.ts
@@ -67,13 +67,11 @@ function createTerminal(command: string, path: string, env: InstallEnv): void {
 }
 
 function escapePath(str: string): string {
-  // on windows, quote everything except the drive letter when the path has one or more spaces
-  // e.g. C:"\Users\user\projects\directory with spaces"
+  // on windows, quote everything except the drive letter (e.g. C:"\Users\user\projects\directory with spaces")
+  // this ensures that all terminals will correctly read the path and will not lose path separators
   if (os.platform() === 'win32') {
-    return str
-      .split(':')
-      .map((partialPath) => (partialPath.includes(' ') ? `"${partialPath}"` : partialPath))
-      .join(':');
+    const [driverLetter, restOfPath] = str.split(':');
+    return `${driverLetter}:"${restOfPath}"`;
   }
 
   return str.replace(/([^A-Za-z0-9_\-.,:/@\n])/g, '\\$1');
@@ -105,7 +103,7 @@ export function generateInstallInfo(
       env['ELECTRON_RUN_AS_NODE'] = 'true';
       command = electronCommand(escapedStorageDir, installLocation);
     } else if (os.platform() === 'win32' && hasCLIBin) {
-      const binPath = join(escapedStorageDir, 'appmap-win-x64.exe');
+      const binPath = escapePath(join(globalStorageDir, 'appmap-win-x64.exe'));
       command = `${binPath} install -d ${installLocation}`;
     }
   }

--- a/src/commands/installAgent.ts
+++ b/src/commands/installAgent.ts
@@ -30,6 +30,17 @@ type InstallInformation = {
   env: InstallEnv;
 };
 
+type TerminalsByOS = {
+  windows: string | null;
+  osx: string | null;
+  linux: string | null;
+};
+
+export type TerminalConfig = {
+  defaultProfile: TerminalsByOS | undefined;
+  shell: TerminalsByOS | undefined;
+};
+
 function getConfigTargetFromConfigLocation(
   configLocation: string
 ): vscode.ConfigurationTarget | undefined {
@@ -114,11 +125,23 @@ export function generateInstallInfo(
   };
 }
 
+function getDefaultTerminals(): TerminalConfig {
+  const terminalConfig = vscode.workspace.getConfiguration('terminal.integrated');
+  const defaultProfile = terminalConfig?.defaultProfile;
+  const shell = terminalConfig?.shell;
+  return {
+    defaultProfile,
+    shell,
+  };
+}
+
 export default async function installAgent(
   context: vscode.ExtensionContext,
   hasCLIBin: boolean
 ): Promise<void> {
   vscode.commands.registerCommand(InstallAgent, async (path: string, language: string) => {
+    const defaultTerminals = getDefaultTerminals();
+
     try {
       const { processService } = context.extension.exports as {
         processService?: NodeProcessService;
@@ -171,16 +194,20 @@ export default async function installAgent(
 
           // restore settings to original state
           await config.update('python.terminal.activateEnvironment', originalValue, configTarget);
-          Telemetry.sendEvent(CLICK_INSTALL_BUTTON, { rootDirectory: path });
+          Telemetry.sendEvent(CLICK_INSTALL_BUTTON, { rootDirectory: path, defaultTerminals });
           return;
         }
       }
 
       createTerminal(command, path, env);
-      Telemetry.sendEvent(CLICK_INSTALL_BUTTON, { rootDirectory: path });
+      Telemetry.sendEvent(CLICK_INSTALL_BUTTON, { rootDirectory: path, defaultTerminals });
     } catch (err) {
       const exception = err as Error;
-      Telemetry.sendEvent(INSTALL_BUTTON_ERROR, { rootDirectory: path, exception });
+      Telemetry.sendEvent(INSTALL_BUTTON_ERROR, {
+        rootDirectory: path,
+        exception,
+        defaultTerminals,
+      });
     }
   });
 }

--- a/src/telemetry/definitions/events.ts
+++ b/src/telemetry/definitions/events.ts
@@ -133,10 +133,14 @@ export const GENERATE_OPENAPI = new Event({
 
 export const CLICK_INSTALL_BUTTON = new Event({
   name: 'install-button:click',
-  properties: [Properties.PROJECT_LANGUAGE],
+  properties: [Properties.PROJECT_LANGUAGE, Properties.DEFAULT_TERMINALS],
 });
 
 export const INSTALL_BUTTON_ERROR = new Event({
   name: 'install-button:error',
-  properties: [Properties.PROJECT_LANGUAGE, Properties.DEBUG_EXCEPTION],
+  properties: [
+    Properties.PROJECT_LANGUAGE,
+    Properties.DEBUG_EXCEPTION,
+    Properties.DEFAULT_TERMINALS,
+  ],
 });

--- a/src/telemetry/definitions/properties.ts
+++ b/src/telemetry/definitions/properties.ts
@@ -7,6 +7,7 @@ import LanguageResolver, { UNKNOWN_LANGUAGE } from '../../services/languageResol
 import { fileExists } from '../../util';
 import ErrorCode from './errorCodes';
 import ProjectMetadata from '../../workspace/projectMetadata';
+import { TerminalConfig } from '../../commands/installAgent';
 
 export const DEBUG_EXCEPTION = new TelemetryDataProvider({
   id: 'appmap.debug.exception',
@@ -165,5 +166,13 @@ export const HAS_INSTALLABLE_PROJECT = new TelemetryDataProvider({
   async value({ projects }: { projects: ProjectMetadata[] }) {
     const isInstallable = projects.some((project) => project?.score && project.score > 1);
     return String(isInstallable);
+  },
+});
+
+export const DEFAULT_TERMINALS = new TelemetryDataProvider({
+  id: 'vscode.workspace.default_terminals',
+  cache: true,
+  async value({ defaultTerminals }: { defaultTerminals: TerminalConfig }) {
+    return JSON.stringify(defaultTerminals);
   },
 });

--- a/test/integration/command/installAgent.test.ts
+++ b/test/integration/command/installAgent.test.ts
@@ -57,7 +57,7 @@ describe('generateInstallInfo function', () => {
       it('generates the correct command for a non-JavaScript project', () => {
         let { command, env } = generateInstallInfo(cwd, 'Ruby', true, globalStorageDir);
         const expected =
-          'C:"\\globalStorageDir\\with spaces in path"\\appmap-win-x64.exe install ' +
+          'C:"\\globalStorageDir\\with spaces in path\\appmap-win-x64.exe" install ' +
           '-d C:"\\Users\\user\\projects\\directory with a space"';
         assert.deepStrictEqual(env, {});
         assert.strictEqual(command, expected);
@@ -95,15 +95,15 @@ describe('generateInstallInfo function', () => {
         assert.deepStrictEqual(env, {});
         assert.strictEqual(
           command,
-          'npx @appland/appmap install -d C:\\Users\\user\\projects\\directory-without-a-space'
+          'npx @appland/appmap install -d C:"\\Users\\user\\projects\\directory-without-a-space"'
         );
       });
 
       it('generates the correct command for a non-JavaScript project', () => {
         let { command, env } = generateInstallInfo(cwd, 'Ruby', true, globalStorageDir);
         const expected =
-          'C:\\globalStorageDir\\without-spaces-in-path\\appmap-win-x64.exe install ' +
-          '-d C:\\Users\\user\\projects\\directory-without-a-space';
+          'C:"\\globalStorageDir\\without-spaces-in-path\\appmap-win-x64.exe" install ' +
+          '-d C:"\\Users\\user\\projects\\directory-without-a-space"';
         assert.deepStrictEqual(env, {});
         assert.strictEqual(command, expected);
 
@@ -117,7 +117,7 @@ describe('generateInstallInfo function', () => {
       it('generates the correct command when there are no CLI binaries', () => {
         let { command, env } = generateInstallInfo(cwd, 'Ruby', false, globalStorageDir);
         const expected =
-          'npx @appland/appmap install -d C:\\Users\\user\\projects\\directory-without-a-space';
+          'npx @appland/appmap install -d C:"\\Users\\user\\projects\\directory-without-a-space"';
         assert.deepStrictEqual(env, {});
         assert.strictEqual(command, expected);
 

